src/hooks/useDebouncedFetchSections.ts

```diff
@@ -55,25 +55,30 @@ const useDebouncedFetchSection = (
   advancedParameters: AdvancedParameters
 ) => {
   const [sectionsData, setSectionsData] = useState<AutocompleteResultSections>({});
+  const [autocompleteParameters, setAutocompleteParameters] =
+    useState<Partial<IAutocompleteParameters>>(advancedParameters);
   const debouncedSearchTerm = useDebounce(query);
 
-  const { numTermsWithGroupSuggestions = 0, numGroupsSuggestedPerTerm = 0 } = advancedParameters;
+  // const { numTermsWithGroupSuggestions = 0, numGroupsSuggestedPerTerm = 0 } = advancedParameters;
+  const numTermsWithGroupSuggestions = 0;
+  const numGroupsSuggestedPerTerm = 0;
 
   useEffect(() => {
```

Comment:
best to avoid managing the same state in multiple locations. Instead, when we can, its always better to derive state. More on this [here](https://dev.to/fpaghar/never-store-derived-state-heres-why-25o6) if you like.

So, rather than managing state lower in the component tree, I bet we can use the useMemo hook here as well.

Nearly everything about this will look the same, except it will be a bit easier to read the src code and there will be minor performance improvements from re-rendering perspective

so no more need for this:
`const [autocompleteParameters, setAutocompleteParameters] =
    useState<Partial<IAutocompleteParameters>>(advancedParameters);`
    
when we do something like this:
```
const decoratedAutocompleteParameters = useMemo(() => {
  const decorated = advancedParameters;
  
   if (autocompleteSections) {
        decorated.resultsPerSection = autocompleteSections.reduce(
            (acc, sectionConfig) => ({
              ...acc,
              [sectionConfig.identifier]: sectionConfig?.numResults || 8,
          }),
        {}),
      }
    }
    
  return decorated;
  }, [autocompleteSections]);
```

Disclaimer. I just wrote this code here in GH comment to demonstrate the pattern as I understand it, but some details may need fine tuning. Don't hesitate to reach out if you'd like to pair on this.